lua/nvim_lsp/mpls.lua

```diff
@@ -0,0 +1,145 @@
+local skeleton = require 'nvim_lsp/skeleton'
+local util = require 'nvim_lsp/util'
+local lsp = vim.lsp
+
+local name = "mpls"
+
+local function file_exists(name)
+   local f=io.open(name,"r")
+   if f~=nil then io.close(f) return true else return false end
+end
+
+local function get_active_interpreter()
+  local f = io.popen("which python | tr -d '\n'") -- runs command
+  local l = f:read("*a") -- read output of command
+  f:close()
+  return l
+end
+
+local function get_python_version()
+  local f = io.popen("python --version 2>&1 | awk '{print $2}' | tr -d '\n'") -- runs command
+  local l = f:read("*a") -- read output of command
+  f:close()
+  return l
+end
+
+local function make_installer()
+  local P = util.path.join
+  local install_dir = P{util.base_install_dir, name}
+
+  local bin = P{install_dir, "Microsoft.Python.LanguageServer.dll"}
+  local cmd = {"dotnet", "exec", bin}
+
+  local X = {}
+  function X.install()
+    local install_info = X.info()
+    if install_info.is_installed then
+      print(name, "is already installed")
+      return
+    end
+    if not (util.has_bins("curl")) then
+      error('Need "curl" to install this.')
+      return
+    end
+    if not util.has_bins("sh", "chmod", "unzip") then
+      error('Need the binaries "sh", "chmod", "unzip" to install this')
```

Comment:
does it really need to be a bash script? `curl` is shipped with Neovim on Windows (plus Windows 10 includes `curl` anyways, these days)

Reply:
Which component? Downloading an extracting the nupkg? Or using Microsoft's bash script to install .net core (Which maybe we should just assume people will handle installing themselves?) I used the sumneko_lua install as inspiration, but I'm happy to eliminate shell commands in favor of doing as much as possible within lua.

Reply:
Everything. I don't see any parts of this PR where a shell is needed. It would be unfortunate if `pyls_ms` didn't actually install for Windows users ...

Reply:
Ok! I'll work on porting it over to lua and adding windows compatibility